### ISSUE TITLE
Breaking change in version 0.25.2 in ExDoc.Autolink module

### ISSUE THREAD

**Issue by svrdlans** [elixir-lang/ex_doc#1395]
Running on Erlang: 24.0.5

In version 0.25.2 there was a change in how a check was made if a module exists. Currently the private function `app_info/1` is used:

```elixir
  defp app_info(module) do
    case :code.which(module) do
      :preloaded ->
        {true, :erts}

      path when is_list(path) ->
        otp? = :string.prefix(path, :code.lib_dir()) != :nomatch

        app =
          case :application.get_application(module) do
            {:ok, app} ->
              app

            _ ->
              case path |> Path.split() |> Enum.reverse() do
                [_, "ebin", app, "lib" | _] ->
                  String.to_atom(app)

                _ ->
                  nil
              end
          end

        {otp?, app}

      :non_existing ->
        {false, nil}
    end
  end
```

It doesn't handle a possible return value of `:code.which/1` - `:cover_compiled`, and in combination with `ex_coveralls` library when there is a ~~behaviour~~ module defined, if it's not tagged with `@moduledoc false`, generating docs will fail with a `CaseClauseError`.

Not sure if this is expected behaviour for `:code.which(module)` returning `:cover_compiled`, when I downgrade ex_doc to 0.25.1 the issue goes away since the implementation is different and it cannot fail for this specific pattern match.

EDIT: set strikethrough on `behaviour` since it's not relevant just to behaviours, I have Ecto schema modules that I have this reproduced.

**Comment by wojtekmach** [comment]
Thank you for the report. Do you have a public project where we could reproduce this? (It's ok if you don't.)

**Comment by svrdlans** [comment]
I don't, but let me prepare one for you.

**Comment by svrdlans** [comment]
The repo is ready [here](https://github.com/svrdlans/simple_ex_docs)

So there is a `bless` mix task created that executes some tasks:
```elixir
    [
      {"compile", ["--warnings-as-errors", "--force"]},
      {"format", ["--check-formatted"]},
      {"coveralls.html", []},
      {"docs", []}
    ]
```
If docs task is invoked after coveralls, then the issue occurs, if I rearrange them, there is no issue. The thing is up until version 0.25.2 there was no issue whatsoever.

**Comment by wojtekmach** [comment]
I'm not familiar with coveralls. Out of curiosity, the module that is reported as :cover_compiled, is it getting docs generated for it? If not, do we expect it to get it? In any case, if it fixes the crash, feel free to send a PR and I'll happily merge it. Again, thank you for reporting it and digging into it!

**Comment by svrdlans** [comment]
Yes, I would like to have docs generated for it, but that's where the CaseClauseError happens. If I disable generating docs by adding `@moduledoc false` after `defmodule`, then the docs are not attempted for that module and there is no issue.
But in my case I wanted to have at least some basic docs in place.

Checking erlang docs for `:code.which/1` it really returns 3 values: `:preloaded`, `:cover_compiled` and `:non_existing`, and I guess that `:cover_compiled` somehow managed not to get triggered, at least until now :)

I'd be happy to send a PR, thank you for responding so quickly.

**Comment by svrdlans** [comment]
When doing changes for the PR and checking Erlang docs again, I noticed that my previous solution wasn't 100% correct, although it did provide working docs.
Better solution would be to handle `:cover_compiled` match separately, because that means that the module was previously found, some stuff were added to it to facilitate counting function invocation for test coverage, and so the solution should be this:
```elixir
      :cover_compiled ->
        {:ok, app} = :application.get_application(module)
        {true, app}
```
We expect the result of `:application.get_application(module)` to always return `{:ok, app}` tuple.

Writing this here for completeness.
